cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go

```diff
@@ -0,0 +1,255 @@
+		Run:          runEtcdLocalPhase,
+		InheritFlags: getControlPlaneJoinPhaseFlags(),
+	}
+}
+
+func newUploadConfigSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "upload-config",
+		Short:        "uploading currently used configuration to the cluster",
+		Run:          runUploadConfigPhaseLocal,
+		InheritFlags: getControlPlaneJoinPhaseFlags(),
+	}
+}
+
+func newMarkControlPlaneSubphases() workflow.Phase {
+	return workflow.Phase{
+		Name:         "mark-control-plane",
+		Short:        "marking the control-plane with right label",
+		Run:          runMarkControlPlanePhaseLocal,
+		InheritFlags: getControlPlaneJoinPhaseFlags(),
+	}
+}
+
+func runEtcdLocalPhase(c workflow.RunData) error {
+	kubeConfigFile := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.AdminKubeConfigFileName)
+
+	client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create Kubernetes client")
+	}
+	data, ok := c.(controlPlaneJoinData)
+	if !ok {
+		return errors.New("etcd phase invoked with an invalid data struct")
+	}
+	cfg, err := data.InitCfg()
+	if err != nil {
+		return err
+	}
+	// in case of local etcd
+	if cfg.Etcd.External != nil {
+		fmt.Println("[control-plane-join] adding etcd in external mode")
+		return nil
+	}
+
+	// Adds a new etcd instance; in order to do this the new etcd instance should be "announced" to
+	// the existing etcd members before being created.
+	// This operation must be executed after kubelet is already started in order to minimize the time
+	// between the new etcd member is announced and the start of the static pod running the new etcd member, because during
+	// this time frame etcd gets temporary not available (only when moving from 1 to 2 members in the etcd cluster).
+	// From https://coreos.com/etcd/docs/latest/v2/runtime-configuration.html
+	// "If you add a new member to a 1-node cluster, the cluster cannot make progress before the new member starts
+	// because it needs two members as majority to agree on the consensus. You will only see this behavior between the time
+	// etcdctl member add informs the cluster about the new member and the new member successfully establishing a connection to the 	// existing one."
+	if err := etcdphase.CreateStackedEtcdStaticPodManifestFile(client, kubeadmconstants.GetStaticPodDirectory(), cfg.NodeRegistration.Name, &cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint); err != nil {
+		return errors.Wrap(err, "error creating local etcd static pod manifest file")
+	}
+
+	return nil
+}
+
+func runUploadConfigPhaseLocal(c workflow.RunData) error {
+	kubeConfigFile := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.AdminKubeConfigFileName)
+
+	client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create Kubernetes client")
+	}
+	data, ok := c.(controlPlaneJoinData)
```

Comment:
same for this function.
data shoul dbe obtained on top.
and data.ClientSetFromFile() should be used.
[1]

Reply:
done